content/docs/08.architecture.md

```diff
@@ -61,26 +61,24 @@ You can scale Executors as necessary, but as no heavy computations are done in t
 
 ### Worker
 
-The **Worker** is where all tasks are run. A Worker will receive a task from the Executor and will process it. As tasks can be nearly anything (heavy computations, simple API calls, etc.), the Worker is essentially a Thread Pool where you can configure the number of threads you need.
+The **Worker** is where all tasks and polling triggers are run. A Worker will receive a task from the Executor or a polling trigger from the Scheduler and will process it. As tasks and polling triggers can be nearly anything (heavy computations, simple API calls, etc.), the Worker is essentially a Thread Pool where you can configure the number of threads you need.
 
 You can scale Workers as necessary and have many instances on multiple servers, each with its own Thread Pool.
 
-As the worker will execute the tasks, it needs access to any external services used by your tasks (database, REST services, message broker, etc.).
+As the worker executes the tasks and the polling triggers, it needs access to any external services they use (database, REST services, message broker, etc.). Workers are the only server components that need access to external services.
 
 ### Scheduler
 
 The **Scheduler** will handle most of the [Triggers](./05.developer-guide/08.triggers/index.md) (except the [Flow Triggers handled by the Executor](./05.developer-guide/08.triggers/02.flow.md)). It will continuously watch all triggers and, if all the conditions are met, will trigger an execution (submit it to the Executor).
 
-Internally, a Scheduler has two Thread Pools:
-- **ScheduledExecutorService**: This Thread Pool will tick every second and try to find an available job.
-- **CachedThreadPool**: This Thread Pool will deal with all other scheduler computational needs: detecting files, executing queries, fetching queues, ...
+In the case of polling triggers, the scheduler will decide if it must be evaluated (based on its evaluation interval), and if it must be, will send the polling trigger to the worker for evaluation. Polling triggers are non-reentrant: they cannot be evaluated concurrently, and they cannot be evaluated if an execution of the flow triggered by this trigger is already running.
```

Comment:
```suggestion
In the case of polling triggers, the scheduler will decide (based on the configured evaluation interval) whether to execute the flow. If the polling trigger conditions are met, the scheduler will send the execution, along with the trigger metadata, to the worker for execution. 

Note that polling triggers cannot be evaluated concurrently. They also can't be reevaluated i.e. if the execution of a flow, which started as a result of the same trigger, is already in a Running state, the next execution will not be started until the previous one finishes.
```
Reentrant is a bit of a confusing/not common word, seems easier to understand without it, up to you

Reply:
Yes, you basically explain what reentrancy is about so OK for me, better to be explicit.